libraries/Microsoft.Bot.Builder.Ai.QnA/QnAMaker.cs

```diff
@@ -129,7 +136,22 @@ public QnAMaker(QnAMakerEndpoint endpoint, QnAMakerOptions options = null, HttpC
                     answer.Score = answer.Score / 100;
                 }
 
-                return results.Answers.Where(answer => answer.Score > _options.ScoreThreshold).ToArray();
+                var result = results.Answers.Where(answer => answer.Score > _options.ScoreThreshold).ToArray();
+
+                var traceInfo = new QnAMakerTraceInfo
+                {
+                    Message = (Activity) messageActivity,
+                    QueryResults = result,
+                    KnowledgeBaseId = _endpoint.KnowledgeBaseId,
+                    ScoreThreshold = _options.ScoreThreshold,
+                    Top = _options.Top,
+                    StrictFilters = _options.StrictFilters,
+                    MetadataBoost = _options.MetadataBoost,
+                };
+                var traceActivity = Activity.CreateTraceActivity(QnAMakerMiddlewareName, QnAMakerTraceType, traceInfo, QnAMakerTraceLabel);
+                await context.SendActivityAsync(traceActivity).ConfigureAwait(false);
+
+                return result;
             }
 
             return null;
```

Comment:
Can this code be refactored so the "main" case of the method doesn't return null? That just seems weird.

Reply:
Heh, good idea.